src/PointwiseFunctions/AnalyticData/GrMhd/MagnetizedFmDisk.hpp

```diff
@@ -0,0 +1,242 @@
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
+#include <limits>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp"
+#include "PointwiseFunctions/GeneralRelativity/KerrSchildCoords.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+// IWYU pragma:  no_include <pup.h>
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+namespace grmhd {
+namespace AnalyticData {
+
+/*!
+ * \brief Magnetized fluid disk orbiting a Kerr black hole.
+ *
+ * In the context of simulating accretion disks, this class implements a widely
+ * used (e.g. \cite Gammie2003, \cite Porth2016rfi, \cite White2015omx)
+ * initial setup for the GRMHD variables, consisting of a Fishbone-Moncrief disk
+ * \cite Fishbone1976apj (see also
+ * RelativisticEuler::Solutions::FishboneMoncriefDisk),
+ * threaded by a weak poloidal magnetic field. The magnetic field is constructed
+ * from an axially symmetric toroidal magnetic potential which, in Kerr
+ * ("spherical Kerr-Schild") coordinates, has the form
+ *
+ * \f{align*}
+ * A_\phi(r,\theta) \propto \text{max}(\rho(r,\theta) - \rho_\text{thresh}, 0),
+ * \f}
+ *
+ * where \f$\rho_\text{thresh}\f$ is a user-specified threshold density that
+ * confines the magnetic flux to exist inside of the fluid disk only. A commonly
+ * used value for this parameter is
+ * \f$\rho_\text{thresh} = 0.2\rho_\text{max}\f$, where \f$\rho_\text{max}\f$
+ * is the maximum value of
+ * the rest mass density in the disk. Using this potential, the Eulerian
+ * magnetic field takes the form
+ *
+ * \f{align*}
+ * B^r = \frac{F_{\theta\phi}}{\sqrt{\gamma}},\quad
+ * B^\theta = \frac{F_{\phi r}}{\sqrt{\gamma}},\quad B^\phi = 0,
+ * \f}
+ *
+ * where \f$F_{ij} = \partial_i A_j - \partial_j A_i\f$ are the spatial
+ * components of the Faraday tensor, and \f$\gamma\f$ is the determinant of the
+ * spatial metric. The magnetic field is then normalized so that the
+ * plasma-\f$\beta\f$ parameter, \f$\beta = 2p/b^2\f$, equals some value
+ * specified by the user. Here, \f$p\f$ is the fluid pressure, and
+ *
+ * \f{align*}
+ * b^2 = b^\mu b_\mu = \frac{B_iB^i}{W^2} + (B^iv_i)^2
+ * \f}
+ *
+ * is the norm of the magnetic field in the fluid frame, with \f$v_i\f$ being
+ * the spatial velocity, and \f$W\f$ the Lorentz factor.
+ */
+class MagnetizedFmDisk {
+ public:
+  using equation_of_state_type = EquationsOfState::PolytropicFluid<true>;
+  using background_spacetime_type = gr::Solutions::KerrSchild;
```

Comment:
Is this needed for anything? I thought we got rid of it.

Reply:
You're right! my bad.